internal/docstore/mongodocstore/mongo.go

```diff
@@ -40,6 +55,89 @@ import (
 	"gocloud.dev/internal/gcerr"
 )
 
+func init() {
+	docstore.DefaultURLMux().RegisterCollection(Scheme, new(defaultDialer))
+}
+
+// defaultDialer dials a default Mongo server based on the environment variable
+// MONGO_SERVER_URL.
+type defaultDialer struct {
+	init   sync.Once
+	opener *URLOpener
+	err    error
+}
+
+func (o *defaultDialer) OpenCollectionURL(ctx context.Context, u *url.URL) (*docstore.Collection, error) {
+	o.init.Do(func() {
+		serverURL := os.Getenv("MONGO_SERVER_URL")
+		if serverURL == "" {
+			o.err = errors.New("MONGO_SERVER_URL environment variable is not set")
+			return
+		}
+		client, err := Dial(ctx, serverURL)
+		if err != nil {
+			o.err = fmt.Errorf("failed to dial default Mongo server at %q: %v", serverURL, err)
+			return
+		}
+		o.opener = &URLOpener{Client: client}
+	})
+	if o.err != nil {
+		return nil, fmt.Errorf("open collection %s: %v", u, o.err)
+	}
+	return o.opener.OpenCollectionURL(ctx, u)
+}
+
+// Dial returns a new mongoDB client that is connected to the server URI.
+func Dial(ctx context.Context, uri string) (*mongo.Client, error) {
+	opts := options.Client().ApplyURI(uri)
+	if err := opts.Validate(); err != nil {
+		return nil, err
+	}
+	client, err := mongo.NewClient(opts)
+	if err != nil {
+		return nil, err
+	}
+	if err := client.Connect(ctx); err != nil {
+		return nil, err
+	}
+	// Connect doesn't seem to actually make a connection, so do an RPC.
+	if err := client.Ping(ctx, nil); err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// Scheme is the URL scheme mongodocstore registers its URLOpener under on
+// docstore.DefaultMux.
+const Scheme = "mongodocstore"
+
+// URLOpener opens URLs like "mongodocstore://mydb/mycollection".
+//
+// The URL Host is used as the database name.
```

Comment:
What do database names look like? E.g., is "foo/bar" a valid database name? If so, this won't work.

Reply:
No, only the collection name can contain `/`s. Added a link to the naming rules.